src/main/java/seedu/address/logic/parser/InternshipParserUtil.java

```diff
@@ -66,19 +99,23 @@ public static Location parseLocation(String location) throws ParseException {
     }
 
     /**
-     * Parses a {@code String description} into a {@code Description}.
+     * Parses a {@code Optional<String> location} into a {@code Location}.
      * Leading and trailing whitespaces will be trimmed.
      *
-     * @throws ParseException if the given {@code description} is invalid.
+     * @throws ParseException if the given {@code location} is invalid.
      */
-    public static Description parseDescription(String description) throws ParseException {
-        requireNonNull(description);
-        String trimmedDescription = description.trim();
-        if (!Description.isValidDescription(trimmedDescription)) {
-            throw new ParseException(Description.MESSAGE_CONSTRAINTS);
+    public static Location parseOptionalLocation(Optional<String> location) throws ParseException {
+        if (location.isEmpty()) {
+            return null;
         }
-        return new Description(trimmedDescription);
-
+        String trimmedLocation = location.get().trim();
+        if (trimmedLocation.isBlank()) {
+            return null;
+        }
+        if (!Location.isValidLocation(trimmedLocation)) {
+            throw new ParseException(Location.MESSAGE_CONSTRAINTS);
+        }
+        return new Location(trimmedLocation);
```

Comment:
Same for the other optional fields. This is to transform them into fields that can accept NULL/blank values, thus making them optional